nucypher/crypto/powers.py

```diff
@@ -123,8 +124,19 @@ def __init__(self,
         self.blockchain = BlockchainInterfaceFactory.get_or_create_interface(provider_uri=provider_uri)
         self.__account = account
 
-        # TODO: Is there a better way to design this Flag? See #1385
-        self.device = True if not password else False
+        # TODO: Temporary fix for #1128 and #1385. It's ugly af, but it works. Move somewhere else?
+        try:
+            wallets = self.blockchain.client.wallets
+        except AttributeError:
+            is_from_hw_wallet = False
+        else:
+            HW_WALLET_URL_PREFIXES = ('trezor', 'ledger')
+            hw_accounts = [w['accounts'] for w in wallets if w['url'].startswith(HW_WALLET_URL_PREFIXES)]
+            hw_addresses = [to_checksum_address(account['address']) for sublist in hw_accounts for account in sublist]
```

Comment:
Almost feels wasteful that we do this cycle for every account. But shouldn't be slow, I guess?

Reply:
It seems too much but in most cases (non hw accounts) these sublists only have one element. They have more than one just when they are hw accounts.